waspc/data/Generator/templates/react-app/src/actions/index.js

```diff
@@ -0,0 +1,74 @@
+import {
+  useMutation,
+  useQueryClient,
+} from 'react-query'
+
+export { configureQueryClient } from '../queryClient'
+
+export function useAction(actionFn, actionOptions) {
+  const queryClient = useQueryClient();
+
+  // TODO(filip): Do we allow all react query options or should we consider a list of allowed options?
+  let options = {}
+
+  if (actionOptions?.optimisticUpdates) {
+    const optimisticUpdatesConfig = makeOptimisticUpdatesConfig(actionOptions.optimisticUpdates)
```

Comment:
We have two optimistic update config "types":
- The public config is the what we document and what our users use
- The internal config is what we use internally (and this is where we remap it)

Currently, the only difference between the two is that the users are allowed to say:
```javascript
getQuery: () => [getTasks.queryCacheKey, taskId]
```
but also:
```javascript
getQuery: () => [getTasks, taskId] // notice the lack of queryCacheKey
```
Our internal config always expects a list of keys, so we have to parse it with the `parseQueryKey` function.

It might make sense to drop the support for one of those definition options (probably the one that requires users explicitly listing `queryCacheKey`, since we're doing our best to abstract it away).

Reply:
@shayneczyzewski said (and I did it, we now only allow functions and still need the mapper):

----

I agree it may make sense just to pick one to support.